install.sh

```diff
@@ -94,133 +103,128 @@ esac
 
 # functions to maintain packages
 function download_check () {
-    cd $CREW_BREW_DIR
+    cd "${CREW_BREW_DIR}"
 
     #download
-    echo "Downloading $1..."
-    curl --progress-bar -C - -# -L --ssl $2 -o "$3"
+    echo "Downloading ${1}..."
+    curl --progress-bar -C - -L --ssl "${2}" -o "${3}"
 
     #verify
-    echo "Verifying $1..."
-    echo $4 $3 | sha256sum -c -
-    case $? in
+    echo "Verifying ${1}..."
+    echo "${4}" "${3}" | sha256sum -c -
+    case "${?}" in
     0) ;;
     *)
-      echo "Verification failed, something may be wrong with the $1 download."
+      echo "Verification failed, something may be wrong with the download."
       exit 1;;
     esac
 }
 
 function extract_install () {
-    cd $CREW_BREW_DIR
+    cd "${CREW_BREW_DIR}/dest"
 
     #extract and install
-    echo "Extracting $1 (this may take a while)..."
-    rm -rf ./usr ./home
-    tar -xf $2
-    echo "Installing $1 (this may take a while)..."
-    if [ -d "${CREW_BREW_DIR}/home" ]; then
-      tar cf - ./usr/* ./home/* | (cd /; tar xp --keep-directory-symlink -f -)
-    else
-      tar cf - ./usr/* | (cd /; tar xp --keep-directory-symlink -f -)
-    fi
-    mv ./dlist $CREW_CONFIG_PATH/meta/$1.directorylist
-    mv ./filelist $CREW_CONFIG_PATH/meta/$1.filelist
+    echo "Extracting ${1} (this may take a while)..."
+    for dir in ./*/; do
+      rm -rf "${dir}"
+    done
+    for file in ./*list; do
+      rm -f "${file}"
+    done
```

Comment:
This might be more thorough and is a bit easier:
```
find . -type d -exec rm -rf {} \\;
find . -type f -exec rm -f {} \\;
```

Reply:
Then try this:
`find -type d -not -path . -exec rm -rf {} \\;`

Reply:
Actually, the more I think about this, the more wrong I think the decision was to change this.

Reply:
Thank you. I'm glad to know my ideas go unappreciated.

Reply:
I'm sorry if I'm being disrespectful, but it just doesn't make me feel happy to work on this when the work that I spend a lot of time on is rejected, as a pose to your contributions, which are committed with almost no question.

Reply:
Nothing is being rejected.  I was saying the idea to delete everything like we're doing here was the wrong idea.  There is stuff in this directory we want to save.

Reply:
Notice that I added `/dest` to the directory we `cd` into. I didn't want the `dest` directory to be removed, but I did want the files to be removed. If this fix isn't incorporated, all the files in `${CREW_PREFIX}/tmp/crew/` stay there throughout the installation. This approach saves space, and is more on flow with what `crew` (the Ruby script) does.

Reply:
We're making this more complicated than needed.  This is all you need:
```
function extract_install () {
    #start with a clean slate
    rm -rf "${CREW_DEST_DIR}"
    mkdir -p "${CREW_DEST_DIR}"
    cd "${CREW_DEST_DIR}"

    #extract and install
    echo "Extracting ${1} (this may take a while)..."
    tar xpf ../"${2}"
    echo "Installing ${1} (this may take a while)..."
    tar cpf - ./*/* | (cd /; tar xp --keep-directory-symlink -f -)
    mv ./dlist "${CREW_CONFIG_PATH}/meta/${1}.directorylist"
    mv ./filelist "${CREW_CONFIG_PATH}/meta/${1}.filelist"
}
```